lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -268,327 +292,167 @@ def main(self):
+                       self.netmasktype, self.device, founddev)
 
-    def set_netmask(self):
-        if self.netmasktype is None:
-            self.set_device()
-
-        device_ip_reg_map = {'eth': "ether.*?\n.*?inet\s+(\d+\.\d+.\d+.\d+/\d+)",
-                             'ib': "infiniband.*?\n.*?inet\s+(\d+\.\d+.\d+.\d+/\d+)"
-                             }
-        if not self.netmasktype in device_ip_reg_map:
-            self.log.raiseException("set_netmask: can't get netmask for %s: unknown mode (device_ip_reg_map %s)" %
-                                    (self.netmasktype, device_ip_reg_map))
-
-        cmd = "/sbin/ip addr show"
-        ec, out = run_simple(cmd)
-        if ec > 0:
-            self.log.raiseException("set_netmask: failed to run cmd %s: %s" % (cmd, out))
-
-        reg = re.compile(r"" + device_ip_reg_map[self.netmasktype])
-        if not reg.search(out):
-            self.log.raiseException("set_netmask: can't get netmask for %s: no matches found (reg %s out %s)" %
-                                    (self.netmasktype, device_ip_reg_map[self.netmasktype], out))
-
-        res = []
-        for ipaddr_mask in reg.finditer(out):
-            ip = IP(ipaddr_mask.group(1), make_net=True)
-            network_netmask = "%s/%s" % (ip.net(), ip.netmask())
-            res.append(network_netmask)
-            self.log.debug("set_netmask: convert ipaddr_mask %s into network_netmask %s" %
-                           (ipaddr_mask.group(1), network_netmask))
-
-        self.log.debug("set_netmask: return complete netmask %s" % res)
-        if len(res) > 0:
-            self.netmask = ":".join(res)
-
-    def make_mympirundir(self):
-        basepath = getattr(self.options, 'basepath', None)
-        if basepath is None:
-            basepath = os.environ['HOME']
-        if not os.path.exists(basepath):
-            self.log.raiseException("make_mympirun_dir: basepath %s should exist." % basepath)
-
-        self.mympirunbasedir = os.path.join(basepath, '.mympirun')
-        destdir = os.path.join(self.mympirunbasedir, "%s_%s" % (self.id, time.strftime("%Y%m%d_%H%M%S")))
-        if not os.path.exists(destdir):
-            try:
-                os.makedirs(destdir)
-            except:
-                self.log.raiseException('make_mympirun_dir: failed to make job dir %s' % (destdir))
-
-        self.log.debug("make_mympirun_dir: tmp mympirundir %s" % destdir)
-        self.mympirundir = destdir
-
-    def get_universe_ncpus(self):
-        """Return ppn for universe"""
-        return self.mpitotalppn
+    def set_device(self, founddev):
+        """Set self.device to founddev, but doublecheck if the path to this device actually exists """
+        self.device = self.DEVICE_MPIDEVICE_MAP[founddev]
+        path = self.DEVICE_LOCATION_MAP[founddev]
+        if path is None or os.path.exists(path):
+            self.log.warning("Forcing device %s (founddevice %s), but path %s not found.",
```

Comment:
but test above is `or  os.path.exists(path)`. should be `or not  os.path.exists(path)`?